### PR TITLE
ts strict part 3: teams

### PR DESCRIPTION
**Pull request by chrisnojima**
Please see https://github.com/keybase/client/pull/18383 for an overview

### CODE REVIEW

File: shared/actions/teams.tsx [hunk 1]
```diff
@@ -529,42 +533,46 @@ function* getDetails(_, action: TeamsGen.GetDetailsPayload, logger) {
...
-        id: invite.id,
-        // @ts-ignore possibly a real issue
-        name: invite.type.c === RPCTypes.TeamInviteCategory.seitan ? invite.name : '',
-        role,
-        username,
-      })
-    }).filter(Boolean)
+        arr.push(
+          Constants.makeInviteInfo({
+            // @ts-ignore TODO this may be a real problem
```
Comment (chrisnojima) [inline shared/actions/teams.tsx]: figured out the issue, our code gen makes the wrong type. ticketed to work on this next

File: shared/teams/team/container.tsx [hunk 2]
```diff
@@ -5,34 +5,30 @@ import Team, {Props} from '.'
...
-  const teamname = getRouteProps(ownProps, 'teamname')
+const mapStateToProps = (state: Container.TypedState, ownProps: OwnProps) => {
+  const teamname = Container.getRouteProps<string>(ownProps, 'teamname')
   if (!teamname) {
     throw new Error('There was a problem loading the team page, please report this error.')
   }
 
   const selectedTab = ownProps.selectedTab || 'members'
 
+  // TODO this shouldn't be implemented as smashing these unrelated states together makes the typing really messy and this hard to understand
```
Comment (chrisnojima) [inline shared/teams/team/container.tsx]: the typing here is nasty as this connector does the work of 2 unrelated things. it really should be split up

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/teams/delete-team/container.tsx [hunk 3]
```diff
@@ -5,12 +5,7 @@ import ReallyDeleteTeam from '.'
 import {deleteTeamWaitingKey} from '../../constants/teams'
 import {anyWaiting} from '../../constants/waiting'
 
-type OwnProps = Container.RouteProps<
-  {
-    teamname: string
-  },
-  {}
->
+type OwnProps = Container.RouteProps< { teamname: string } >
```
Comment (cjb) [inline shared/teams/delete-team/container.tsx]: seems like `type OwnProps = Container.RouteProps<{teamname: string}>` is more our style?

Comment (chrisnojima) [reply shared/teams/delete-team/container.tsx]: fixed tnx

**Pull request merged**